### PR TITLE
Add max idle time as a configurable property for the mongodb connector

### PR DESCRIPTION
**Pull request by jerroldlaw**
There is an issue present when connecting to MongoDB hosted in Azure Cosmos DB.

Connections made to Cosmos DB are dropped after 10 minutes without any communication back to the client. This causes Presto to fail queries with a timeout exception during queries to the mongo catalog when idle for 10 minutes after the previously succeeded query.

In Cosmos DB, the default connection string given to users has a config `maxIdleTimeMS=120000`. Allowing this property in the Presto mongodb connector will close connections after the configured time. This will prevent the timeout exceptions from happening.

This PR adds the maxConnectionIdleTime to the list of configurable properties in the mongodb connector.

### CODE REVIEW

File: presto-mongodb/src/main/java/io/prestosql/plugin/mongodb/MongoClientModule.java [hunk 1]
```diff
@@ -56,7 +56,8 @@ public static MongoSession createMongoSession(TypeManager typeManager, MongoClie
                 .maxWaitTime(config.getMaxWaitTime())
                 .minConnectionsPerHost(config.getMinConnectionsPerHost())
                 .readPreference(config.getReadPreference().getReadPreference())
-                .writeConcern(config.getWriteConcern().getWriteConcern());
+                .writeConcern(config.getWriteConcern().getWriteConcern())
+                .maxConnectionIdleTime(config.getMaxConnectionIdleTime());
```
[the remark this answers is not in the document]
Comment (jerroldlaw) [reply presto-mongodb/src/main/java/io/prestosql/plugin/mongodb/MongoClientModule.java]: Done

File: presto-mongodb/src/main/java/io/prestosql/plugin/mongodb/MongoClientConfig.java [hunk 2]
```diff
@@ -307,4 +308,16 @@ public MongoClientConfig setSslEnabled(boolean sslEnabled)
         this.sslEnabled = sslEnabled;
         return this;
     }
+
+    public int getMaxConnectionIdleTime()
```
[the remark this answers is not in the document]
Comment (jerroldlaw) [reply presto-mongodb/src/main/java/io/prestosql/plugin/mongodb/MongoClientConfig.java:313]: Added the annotation

File: presto-mongodb/src/main/java/io/prestosql/plugin/mongodb/MongoClientConfig.java [hunk 3]
```diff
@@ -307,4 +308,16 @@ public MongoClientConfig setSslEnabled(boolean sslEnabled)
...
+    {
+        return maxConnectionIdleTime;
+    }
+
+    @Config("mongodb.max-connection-idle-time-ms")
```
[the remark this answers is not in the document]
Comment (jerroldlaw) [reply presto-mongodb/src/main/java/io/prestosql/plugin/mongodb/MongoClientConfig.java]: Yup agreed. Removed the "-ms" for more consistency with the rest of the configs. Thanks for taking time to review this @ebyhr 😄

[the remark this answers is not in the document]
Comment (jerroldlaw) [reply presto-mongodb/src/main/java/io/prestosql/plugin/mongodb/MongoClientConfig.java]: Updated `mongodb.rst` too.

**Comment by ebyhr** [comment]
Merged, thanks!

**Pull request merged**